### PR TITLE
Trying out clang-format 5.0.0

### PR DESCRIPTION
**Pull request by dkorolev**
@mzhurovich, what do you think of `clang-format-5`?

### CODE REVIEW

**Comment by dkorolev** [comment]
@mzhurovich, this one may be worth merging before C5T's https://github.com/C5T/Current/pull/785 is in.

File: bricks/strings/chunk.h [hunk 1]
```diff
@@ -190,12 +189,12 @@ class UniqueChunk final : public Chunk {
 #define DEFINE_COMPARATOR(OP) \
   bool operator OP(const UniqueChunk& rhs) const { return S OP rhs.S; }
   // It's `clang-format` inserting those spaces between the operator and closing parenthesis; not me. -- D.K.
```
Comment (mzhurovich) [inline bricks/strings/chunk.h:191]: Now obsolete :)

File: bricks/sync/waitable_atomic.h [hunk 2]
```diff
@@ -149,9 +147,8 @@ class WaitableAtomicImpl {
      public:
       using parent_t = PARENT;
       using optional_notifier_t = typename NotifyIfMutable<PARENT>::impl_t;
-      using data_t = typename std::conditional<std::is_const<parent_t>::value,
-                                               const typename parent_t::data_t,
-                                               typename parent_t::data_t>::type;
+      using data_t = typename std::
+          conditional<std::is_const<parent_t>::value, const typename parent_t::data_t, typename parent_t::data_t>::type;
```
Comment (mzhurovich) [inline bricks/sync/waitable_atomic.h:151]: That's a bit weird :-/

File: bricks/util/object_interface.h [hunk 3]
```diff
@@ -41,12 +41,12 @@ SOFTWARE.
...
 
-#define CURRENT_FORWARD_METHOD(method)                                                                                \
-  template <typename... ARGS>                                                                                         \
-  typename std::result_of<decltype(&CURRENT_IMPLEMENTATION_POINTER_TYPE::method)(CURRENT_IMPLEMENTATION_POINTER_TYPE, \
-                                                                                 ARGS...)>::type                      \
-  method(ARGS&&... args) {                                                                                            \
-    return this->CURRENT_IMPLEMENTATION_POINTER->method(std::forward<ARGS>(args)...);                                 \
+#define CURRENT_FORWARD_METHOD(method)                                                                                 \
+  template <typename... ARGS>                                                                                          \
+  typename std::result_of<decltype (&CURRENT_IMPLEMENTATION_POINTER_TYPE::method)(CURRENT_IMPLEMENTATION_POINTER_TYPE, \
```
Comment (mzhurovich) [inline bricks/util/object_interface.h:46]: wut?

File: bricks/util/test.cc [hunk 4]
```diff
@@ -839,7 +833,8 @@ TEST(ObjectInterface, Smoke) {
   }
   {
     object_with_interface_test_namespace::NamespacedExampleObjectInterface<
-        object_with_interface_test_namespace::NamespacedObjectWithInterface> o;
+        object_with_interface_test_namespace::NamespacedObjectWithInterface>
+        o;
```
Comment (mzhurovich) [inline bricks/util/test.cc:837]: o_O

File: compact_tsv/test.cc [hunk 5]
```diff
@@ -116,12 +128,12 @@ TEST(CompactTSV, Smoke) {
...
-    std::cerr << "Unpack into std::ostringstream:             " << K *(t_c_end - t_c_begin).count() << "ms.\n";
-    std::cerr << "Unpack into std::string-s:                  " << K *(t_d_end - t_d_begin).count() << "ms.\n";
-    std::cerr << "Unpack into std::pair<const char*, size_t>: " << K *(t_e_end - t_e_begin).count() << "ms.\n";
-    std::cerr << "Unpack into UniqueChunk-s:                  " << K *(t_f_end - t_f_begin).count() << "ms.\n";
+    std::cerr << "Generate:                                   " << K * (t_a_end - t_a_begin).count() << "ms.\n";
+    std::cerr << "Pack:                                       " << K * (t_b_end - t_b_begin).count() << "ms.\n";
+    std::cerr << "Unpack into std::ostringstream:             " << K * (t_c_end - t_c_begin).count() << "ms.\n";
+    std::cerr << "Unpack into std::string-s:                  " << K * (t_d_end - t_d_begin).count() << "ms.\n";
+    std::cerr << "Unpack into std::pair<const char*, size_t>: " << K * (t_e_end - t_e_begin).count() << "ms.\n";
+    std::cerr << "Unpack into UniqueChunk-s:                  " << K * (t_f_end - t_f_begin).count() << "ms.\n";
```
Comment (mzhurovich) [inline compact_tsv/test.cc:136]: 👍

File: examples/benchmark/generic/scenario_storage.h [hunk 6]
```diff
@@ -91,32 +91,34 @@ SCENARIO(storage, "Storage transactions test.") {
...
-               std::exit(-1);
-             }
-           }).Wait();
+               if (fields.hashmap_uint32.Size() != actual_size_uint32 ||
+                   fields.hashmap_string.Size() != actual_size_string) {
+                 std::cerr << "Test failed." << std::endl;
+                 std::exit(-1);
+               }
+             })
+               .Wait();
```
Comment (mzhurovich) [inline examples/benchmark/generic/scenario_storage.h:100]: I don't like it :(

File: examples/type_evolution/test.cc [hunk 7]
```diff
@@ -37,25 +37,21 @@ CURRENT_STRUCT_EVOLVER(CustomEvolver, SchemaFrom, FullName, into.full_name = fro
...
-                       {
-                         typename INTO::CustomTypeA value;
-                         value.a = from.c + 1;
-                         into = std::move(value);
-                       });
+  CURRENT_EVOLVE_CASE(CustomTypeC, {
+    typename INTO::CustomTypeA value;
+    value.a = from.c + 1;
+    into = std::move(value);
+  });
```
Comment (mzhurovich) [inline examples/type_evolution/test.cc:44]: 👍

File: karl/karl.h [hunk 8]
```diff
@@ -314,33 +316,37 @@ class GenericKarl final : private KarlStorage<STORAGE_TYPE>,
...
-      }
-      fields.karl.Add(self_info);
+    storage_
+        ->ReadWriteTransaction([this](MutableFields<storage_t> fields) {
+          const auto& keepalives_data = keepalives_stream_->Data();
+          KarlInfo self_info;
+          if (!keepalives_data->Empty()) {
+            self_info.persisted_keepalives_info = keepalives_data->LastPublishedIndexAndTimestamp();
+          }
+          fields.karl.Add(self_info);
```
Comment (mzhurovich) [inline karl/karl.h:326]: And this one became worse :(

**Review by mzhurovich — CHANGES_REQUESTED** [review]
I have mixed feelings about this. clang-format 5 is definitely better in dealing with lambdas, but some cases looks really weird for me :(
From the offline chat - we can definitely think about setting
```
SortIncludes=false # Don't rearrange headers
FixNamespaceComments=false # Don't "fix" our extended namespace comments like `//  blah::impl`
```

**Pull request closed without merging**